From c404bc69c339d088a0c556667681f83a3c6e06b4 Mon Sep 17 00:00:00 2001
From: Bijoy Jose <bijoy.a.jose@intel.com>
Date: Thu, 3 Jan 2013 15:32:38 -0800
Subject: Dalivk: fix for Compiler, add OP_SGET/SPUT_OBJECT_VOLATILE opcode in assert

BZ: 75820

op_sget and op_sput functions miss the OP_SGET_OBJECT_VOLATILE and
OP_SPUT_OBJECT_VOLATILE opcodes in assert statement.
That results in assert is thrown in case the corresponding opcode is used.
The patch adds the mentioned opcodes to assert statement.
The callers of mentioned functions accept these opcodes.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: Ifda6e55856d9124caa92474c4f92fa2540d65fad
Orig-MCG-Change-Id: If994b9f8c4ce6a07637c9671e079182a24652586
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/codegen/x86/LowerGetPut.cpp |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/vm/compiler/codegen/x86/LowerGetPut.cpp b/vm/compiler/codegen/x86/LowerGetPut.cpp
index 1a94b8d..9da7420 100644
--- a/vm/compiler/codegen/x86/LowerGetPut.cpp
+++ b/vm/compiler/codegen/x86/LowerGetPut.cpp
@@ -986,7 +986,7 @@ int sget_sput_common(StaticAccess flag, u2 vA, u2 referenceIndex, bool isObj,
 
 /**
  * @brief Generate native code for bytecodes sget, sget-boolean,
- * sget-byte, sget-char, sget-object, sget-short, and sget/volatile
+ * sget-byte, sget-char, sget-object, sget-short, sget/volatile and sget-object/volatile
  * @param mir bytecode representation
  * @return value >= 0 when handled
  */
@@ -997,7 +997,8 @@ int op_sget(const MIR * mir) {
             || mir->dalvikInsn.opcode == OP_SGET_CHAR
             || mir->dalvikInsn.opcode == OP_SGET_OBJECT
             || mir->dalvikInsn.opcode == OP_SGET_SHORT
-            || mir->dalvikInsn.opcode == OP_SGET_VOLATILE);
+            || mir->dalvikInsn.opcode == OP_SGET_VOLATILE
+            || mir->dalvikInsn.opcode == OP_SGET_OBJECT_VOLATILE);
     u2 vA = mir->dalvikInsn.vA;
     u2 referenceIndex = mir->dalvikInsn.vB;
     int retval = sget_sput_common(SGET, vA, referenceIndex, false, false, mir);
@@ -1074,7 +1075,7 @@ int op_sget_short(const MIR * mir) {
 
 /**
  * @brief Generate native code for bytecodes sput, sput-boolean,
- * sput-byte, sput-char, sput-object, sput-short, and sput/volatile
+ * sput-byte, sput-char, sput-object, sput-short, sput/volatile and sput-object/volatile
  * @param mir bytecode representation
  * @return value >= 0 when handled
  */
@@ -1085,7 +1086,8 @@ int op_sput(const MIR * mir, bool isObj) {
             || mir->dalvikInsn.opcode == OP_SPUT_CHAR
             || mir->dalvikInsn.opcode == OP_SPUT_OBJECT
             || mir->dalvikInsn.opcode == OP_SPUT_SHORT
-            || mir->dalvikInsn.opcode == OP_SPUT_VOLATILE);
+            || mir->dalvikInsn.opcode == OP_SPUT_VOLATILE
+            || mir->dalvikInsn.opcode == OP_SPUT_OBJECT_VOLATILE);
     u2 vA = mir->dalvikInsn.vA;
     u2 referenceIndex = mir->dalvikInsn.vB;
     int retval = sget_sput_common(SPUT, vA, referenceIndex, isObj, false, mir);
-- 
1.7.4.1

